spec/jobs/reports/drop_off_report_spec.rb

```diff
@@ -7,11 +7,62 @@
     JSON.parse '[{"emails":["ursula@example.com"],
        "issuers":["urn:gov:gsa:openidconnect.profiles:sp:sso:agency_name:app_name"]}]'
   end
+  let(:empty_emailable_report) do
+    report = Reporting::DropOffReport.new(
+      issuers: 'urn:gov:gsa:openidconnect.profiles:sp:sso:agency_name:app_name',
+      time_range: report_date.all_week,
+      slice: 1.week,
+    )
+
+    cloudwatch_client = instance_double(Reporting::CloudwatchClient)
+    allow(Reporting::CloudwatchClient).to receive(:new).and_return(cloudwatch_client)
+    allow(cloudwatch_client).to receive(:fetch).with(
+      from: report_date.beginning_of_week,
+      query: a_string_including('coalesce(properties.event_properties.success, 0) AS success'),
+      to: report_date.end_of_week,
+    ).and_return([])
+
+    report.as_emailable_reports
+  end
 
   before do
     allow(IdentityConfig.store).to receive(:drop_off_report_config).and_return(report_config)
   end
 
+  describe 'as an integrated feature' do
+    before do
+      cloudwatch_client = instance_double(Reporting::CloudwatchClient)
+      expect(Reporting::CloudwatchClient).to receive(:new).and_return(cloudwatch_client)
+      expect(cloudwatch_client).to receive(:fetch).with(
+        from: report_date.beginning_of_week,
+        query: a_string_including('coalesce(properties.event_properties.success, 0) AS success'),
+        to: report_date.end_of_week,
+      ).and_return([])
+    end
```

Comment:
I would recommend against stubbing like this

See this [PR where I undid a bunch of stubbing of our cloudwatch client](https://github.com/18F/identity-idp/pull/10875/files#diff-a459e66df4727e56629db8e1399de87cf7b5bc44494fe0f7bc11530208ea58c8R76) and added a helper that stubs responses from cloudwatch directly (more like webmock)

See `stub_cloudwatch_responses` and `stub_multiple_cloudwatch_responses`, that way we don't have to stub `Reporting::DropOffReport` or `Reporting::CloudwatchClient` and this test can "go all the way" and be a more realistic test

Reply:
Thank you! I definitely feel better about the tests writing them that way